bot/command_error_handlers/check_failure.py

```diff
@@ -0,0 +1,40 @@
+from typing import NoReturn
+
+from discord import Embed, Interaction
+from discord.ext.commands import CheckFailure, Context, NoPrivateMessage
+from pydis_core.utils.error_handling.commands import AbstractCommandErrorHandler
+
+from bot.constants import Channels, NEGATIVE_REPLIES
+from bot.utils.decorators import InChannelCheckFailure, InMonthCheckFailure
+
+from ._utils import create_error_embed
+
+
+class CheckFailureErrorHandler(AbstractCommandErrorHandler):
+    """An handler for the CheckFailure error."""
+
+    async def should_handle_error(self, error: Exception) -> bool:
+        """A predicate that determines whether the error should be handled or not."""
+        return isinstance(error, CheckFailure)
+
+    async def handle_text_command_error(self, context: Context, error: Exception) -> NoReturn:
```

Comment:
The default value of a function is `None`. If a function can return, but has no explicit return value, then this annotation should be `None`. 

`NoReturn`  means it does not return i.e. control flow of the program is interrupted before it returns. I've only seen this used for functions that always throw an exception, like an abstract method that raises NotImplemented. Also, this was superceded (?) in 3.11 by `Never`.